packages/plugins/other/visitor-plugin-common/src/client-side-base-visitor.ts

```diff
@@ -346,6 +358,59 @@ export class ClientSideBaseVisitor<
     return documentStr;
   }
 
+  private _generateDocumentNodeHash(
```

Comment:
To me, it feels like this function shouldn't be here at all. Given the way how different document nodes are constructed (see `_gql` method), I don't see any other way right now...